Zend/zend.c

```diff
@@ -1122,6 +1128,9 @@ void zend_shutdown(void) /* {{{ */
 	zend_vm_dtor();
 
 	zend_destroy_rsrc_list(&EG(persistent_list));
+#ifdef ZTS
+	ts_callback_id(executor_globals_id, executor_globals_persistent_list_dtor);
+#endif
```

Comment:
This probably may be changed into (I'm not completely sure)
```c

#ifndef ZTS
	zend_destroy_rsrc_list(&EG(persistent_list));
#else
	ts_callback_id(executor_globals_id, executor_globals_persistent_list_dtor);
#endif
```

Reply:
It also may make sense to move this logic into ``module_destructor()``.

Reply:
I just quickly tested this by printing out all the lists that get destroyed, and I see that it now misses the `global_persistent_list` whereas it previously was destroyed by `zend_destroy_rsrc_list(&EG(persistent_list));`. So this seems wrong.